openwisp_radius/migrations/0043_alter_organizationradiussettings_sms_meta_data_and_more.py

```diff
@@ -0,0 +1,52 @@
+# Generated by Django 5.2.5 on 2025-10-22 18:34
+
+import json
+
+from django.db import migrations, models
+
+
+def convert_user_credentials_data(apps, schema_editor):
+    """
+    Convert existing double-encoded JSON strings in user_credentials field
+    to proper JSON objects for Django's built-in JSONField.
+    """
+    RadiusBatch = apps.get_model("openwisp_radius", "RadiusBatch")
+    for batch in RadiusBatch.objects.exclude(user_credentials__isnull=True):
+        if isinstance(batch.user_credentials, str):
+            try:
+                batch.user_credentials = json.loads(batch.user_credentials)
+                batch.save(update_fields=["user_credentials"])
+            except (json.JSONDecodeError, TypeError):
+                # If it's already decoded or invalid JSON, skip it
```

Comment:
It shouldn't fail. If it fails we must let it fail loudly, unless there can be failure which would be common and in that case we should catch the exception and resolve this automatically.

Why do you think it would fail here?